roles/contiv_network/tasks/main.yml

```diff
@@ -1,44 +1,13 @@
 ---
 # This role contains tasks for configuring and starting netmaster and netplugin service
+- name: check dns container image
+  shell: docker inspect skynetservices/skydns
+  register: docker_inspect_result
+  ignore_errors: yes
 
-- name: download netmaster and netplugin
-  get_url:
-    validate_certs: no
-    url: https://github.com/contiv/netplugin/releases/download/v0.0.0-12-11-2015.20-54-40.UTC/netplugin-v0.0.0-12-11-2015.20-54-40.UTC.tar.bz2
-    dest: /tmp/contivnet.tar.bz2
+- name: pull dns container image
+  shell: docker pull skynetservices/skydns
+  when: "'No such image' in docker_inspect_result.stderr"
```

Comment:
sure, sgtm. I was hoping we could depend on docker daemon to optimize that for us i.e. not pull an image if it already exists on the machine. I tried looking around docker code and it does appears that it tries to reach out to registry for every pull, so may be this optimization is good.